kubetest2/kubetest2-gce/deployer/up.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog"
+	"k8s.io/test-infra/kubetest2/pkg/exec"
+)
+
+func (d *deployer) Up() error {
+	klog.Info("GCE deployer starting Up()")
+
+	if err := d.verifyFlags(); err != nil {
+		return fmt.Errorf("up could not verify flags: %s", err)
+	}
+
+	if err := enableComputeAPI(d.GCPProject); err != nil {
+		return fmt.Errorf("up couldn't enable compute API: %s", err)
+	}
+
+	env := d.buildEnv()
+	script := filepath.Join(d.RepoRoot, "cluster", "kube-up.sh")
+	klog.Infof("About to run script at: %s", script)
+	klog.Infof("Running under environment: %+v", env)
+
+	cmd := exec.Command(script)
+	cmd.SetEnv(env...)
+	exec.InheritOutput(cmd)
+	err := cmd.Run()
+	if err != nil {
+		return fmt.Errorf("error encountered during %s: %s", script, err)
+	}
+
+	return nil
+}
+
+func (d *deployer) buildEnv() []string {
+	// The base env currently does not inherit the current os env (except for PATH)
+	// because (for now) it doesn't have to. In future, this may have to change when
+	// support is added for k/k's kube-up.sh and kube-down.sh which support a wide
+	// variety of environment variables. Before doing so, it is worth investigating
+	// inheriting the os env vs. adding flags to this deployer on a case-by-case
+	// basis to support individual environment configurations.
+	env := []string{}
```

Comment:
its up to you, but I typically see 
`var env []string` instead of initializing the slice

Reply:
Sure.